### PR TITLE
Brandon createpresenter14

### PR DESCRIPTION
**Pull request by branflake2267**
progress report... working my way through the create presenter steps.

### CODE REVIEW

File: src/com/arcbees/plugin/idea/wizards/createpresenter/CreatePresenterAction.java [hunk 1]
```diff
@@ -123,6 +142,120 @@ private void run() {
...
     }
 
+    /**
+     * create name tokens class, if it doesn't exist
+     */
+    private void createNameTokensFieldAndMethods() {
+        if (!presenterConfigModel.isUsePlace()) {
+            return;
+        }
+        PsiClass unitNameTokens = presenterConfigModel.getNameTokenPsiClass();
```
Comment (meriouma) [inline src/com/arcbees/plugin/idea/wizards/createpresenter/CreatePresenterAction.java:522]: nl

File: src/com/arcbees/plugin/idea/domain/PresenterConfigModel.java [hunk 2]
```diff
@@ -298,4 +298,33 @@ public void setNameTokenPsiClass(PsiClass nameTokenPsiClass) {
...
+        }
+        String s = nameTokenPsiClass.getName().replace(".java", "") + "." + nameToken;
+        return s;
+    }
+
+    public String getNameTokenUnitImport() {
+        if (nameTokenPsiClass == null) {
+            return "";
+        }
+        String importString = "import " + nameTokenPsiClass.getQualifiedName() + ";";
```
Comment (meriouma) [inline src/com/arcbees/plugin/idea/domain/PresenterConfigModel.java]: nl and return directly

File: src/com/arcbees/plugin/idea/domain/PresenterConfigModel.java [hunk 3]
```diff
@@ -298,4 +298,33 @@ public void setNameTokenPsiClass(PsiClass nameTokenPsiClass) {
...
+
+    public String getSelectedPackageAndNameAsSubPackage() {
+        return selectedPackage.getQualifiedName() + "." + getName().toLowerCase();
+    }
+
+    public String getNameTokenWithClass() {
+        if (nameTokenPsiClass == null) {
+            return "";
+        }
+        String s = nameTokenPsiClass.getName().replace(".java", "") + "." + nameToken;
```
Comment (meriouma) [inline src/com/arcbees/plugin/idea/domain/PresenterConfigModel.java]: nl and return directly

File: src/com/arcbees/plugin/idea/domain/PresenterConfigModel.java [hunk 4]
```diff
@@ -298,4 +298,33 @@ public void setNameTokenPsiClass(PsiClass nameTokenPsiClass) {
...
+        String importString = "import " + nameTokenPsiClass.getQualifiedName() + ";";
+        return importString;
+    }
+
+    public String getContentSlotImport() {
+        if (contentSlot == null) {
+            return "";
+        }
+        // TODO ?
+        String importString = "import " + getContentSlot() + ";";
```
Comment (meriouma) [inline src/com/arcbees/plugin/idea/domain/PresenterConfigModel.java]: nl and return directly

File: src/com/arcbees/plugin/idea/wizards/createpresenter/CreatePresenterAction.java [hunk 5]
```diff
@@ -123,6 +142,120 @@ private void run() {
...
+
+    private void fetchPopupPresenterTemplate(PresenterOptions presenterOptions) throws Exception {
+        PopupPresenterOptions presenterWidgetOptions = new PopupPresenterOptions();
+        presenterWidgetOptions.setSingleton(presenterConfigModel.isUseSingleton());
+        presenterWidgetOptions.setCustom(presenterConfigModel.isUseOverrideDefaultPopup());
+
+        try {
+            createdPopupPresenterTemplates = CreatePopupPresenter.run(presenterOptions, presenterWidgetOptions, true);
+        } catch (Exception e) {
+            throw e;
```
Comment (meriouma) [inline src/com/arcbees/plugin/idea/wizards/createpresenter/CreatePresenterAction.java]: why do you catch it?

File: src/com/arcbees/plugin/idea/wizards/createpresenter/CreatePresenterAction.java [hunk 6]
```diff
@@ -123,6 +142,120 @@ private void run() {
...
+    }
+
+    private void fetchPresenterWidgetTemplate(PresenterOptions presenterOptions) throws Exception {
+        PresenterWidgetOptions presenterWidgetOptions = new PresenterWidgetOptions();
+        presenterWidgetOptions.setSingleton(presenterConfigModel.isUseSingleton());
+
+        try {
+            createdPresenterWidgetTemplates = CreatePresenterWidget.run(presenterOptions, presenterWidgetOptions, true);
+        } catch (Exception e) {
+            throw e;
```
Comment (meriouma) [inline src/com/arcbees/plugin/idea/wizards/createpresenter/CreatePresenterAction.java]: Catch?

File: src/com/arcbees/plugin/idea/wizards/createpresenter/CreatePresenterAction.java [hunk 7]
```diff
@@ -123,6 +142,120 @@ private void run() {
...
+        } else if (presenterConfigModel.getPopupPresenter()) {
+            nestedPresenterOptions.setRevealType("RootPopup");
+        } else if (presenterConfigModel.getRevealInSlot()) {
+            nestedPresenterOptions.setRevealType(presenterConfigModel.getContentSlot());
+        }
+
+        try {
+            createdNestedPresenterTemplates = CreateNestedPresenter.run(presenterOptions, nestedPresenterOptions, true);
+        } catch (Exception e) {
+            throw e;
```
Comment (meriouma) [inline src/com/arcbees/plugin/idea/wizards/createpresenter/CreatePresenterAction.java]: catch

File: src/com/arcbees/plugin/idea/wizards/createpresenter/CreatePresenterAction.java (hunk 5 above)
Comment (branflake2267) [inline src/com/arcbees/plugin/idea/wizards/createpresenter/CreatePresenterAction.java]: good point :)

**Pull request merged**